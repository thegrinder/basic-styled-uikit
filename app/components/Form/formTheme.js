const formTheme = {
  // text input, textarea, and select styles
  regular: {
    valid: {
      normal: {
        borderColor: '#e5e5e5',
        color: '#666',
      },
      active: {
        borderColor: '#1e87f0',
        color: '#666',
      },
      disabled: {
        backgroundColor: '#f8f8f8',
        color: '#999',
        borderColor: '#e5e5e5',
      },
    },
    invalid: {
      color: '#f0506e',
      borderColor: '#f0506e',
    },
  },
  // radio and checkbox styles
  misc: {
    normal: {
      borderColor: '#ccc',
      backgroundColor: 'transparent',
    },
    active: {
      borderColor: 'transparent',
      backgroundColor: '#1e87f0',
    },
    disabled: {
      bordeColor: '#e5e5e5',
      backgroundColor: '#f8f8f8',
    },
  },
};

export default formTheme;
