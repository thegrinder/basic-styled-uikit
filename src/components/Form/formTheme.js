const formTheme = {
  // text input, textarea, and select styles
  regular: {
    valid: {
      normal: {
        backgroundColor: '#fff',
        borderColor: '#e5e5e5',
        color: '#666',
      },
      active: {
        backgroundColor: '#fff',
        borderColor: '#1e87f0',
        color: '#666',
      },
      disabled: {
        backgroundColor: '#f8f8f8',
        borderColor: '#e5e5e5',
        color: '#999',
      },
    },
    invalid: {
      backgroundColor: '#fff',
      color: '#f0506e',
      borderColor: '#f0506e',
    },
  },
  // inputs for inline editing
  inline: {
    valid: {
      normal: {
        backgroundColor: 'transparent',
        borderColor: 'transparent',
        color: '#666',
      },
      hover: {
        backgroundColor: 'transparent',
        borderColor: '#e5e5e5',
        color: '#666',
      },
      active: {
        backgroundColor: 'transparent',
        borderColor: '#1e87f0',
        color: '#666',
      },
      disabled: {
        backgroundColor: '#f8f8f8',
        borderColor: '#e5e5e5',
        color: '#999',
      },
    },
    invalid: {
      backgroundColor: 'transparent',
      color: '#f0506e',
      borderColor: '#f0506e',
    },
  },
  // radio and checkbox styles
  misc: {
    valid: {
      normal: {
        borderColor: '#ccc',
        backgroundColor: 'transparent',
      },
      active: {
        borderColor: '#1e87f0',
        backgroundColor: 'transparent',
      },
      checked: {
        borderColor: 'transparent',
        backgroundColor: '#1e87f0',
      },
      disabled: {
        bordeColor: '#e5e5e5',
        backgroundColor: '#f8f8f8',
      },
    },
    invalid: {
      borderColor: '#f0506e',
      backgroundColor: 'transparent',
    },
  },
  toggle: {
    normal: {
      backgroundColor: '#ccc',
    },
    checked: {
      backgroundColor: '#1e87f0',
    },
  },
};

export default formTheme;
