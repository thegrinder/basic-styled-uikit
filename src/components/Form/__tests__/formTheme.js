import formTheme from '../formTheme';

describe('formTheme', () => {
  it('should be defined', () => {
    expect(formTheme).toBeDefined();
    expect(formTheme).toMatchSnapshot();
  });
});
