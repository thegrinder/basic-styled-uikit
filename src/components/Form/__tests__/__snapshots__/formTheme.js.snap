// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formTheme should be defined 1`] = `
Object {
  "inline": Object {
    "common": Object {
      "fontFamily": "Lato, Arial, Helvetica, sans-serif",
      "fontSize": "1rem",
    },
    "dark": Object {
      "states": Object {
        "invalid": Object {
          "borderColor": "#F4511E",
          "color": "#F4511E",
        },
        "valid": Object {
          "active": Object {
            "borderColor": "#1E88E5",
          },
          "disabled": Object {
            "borderColor": "rgba(206,212,218,0.2)",
            "color": "rgba(255,255,255,0.2)",
          },
          "hover": Object {
            "borderColor": "#dee2e6",
          },
          "normal": Object {
            "color": "#fff",
          },
        },
      },
    },
    "light": Object {
      "states": Object {
        "invalid": Object {
          "borderColor": "#F4511E",
          "color": "#F4511E",
        },
        "valid": Object {
          "active": Object {
            "borderColor": "#1E88E5",
            "color": "#495057",
          },
          "disabled": Object {
            "backgroundColor": "#f8f9fa",
            "borderColor": "#dee2e6",
            "color": "#868e96",
          },
          "hover": Object {
            "borderColor": "#dee2e6",
            "color": "#495057",
          },
          "normal": Object {
            "color": "#495057",
          },
        },
      },
    },
  },
  "misc": Object {
    "common": Object {},
    "states": Object {
      "invalid": Object {
        "borderColor": "#F4511E",
      },
      "valid": Object {
        "active": Object {
          "borderColor": "#1E88E5",
        },
        "checked": Object {
          "backgroundColor": "#1E88E5",
          "borderColor": undefined,
        },
        "normal": Object {
          "borderColor": "#dee2e6",
        },
      },
    },
  },
  "regular": Object {
    "common": Object {
      "fontFamily": "Lato, Arial, Helvetica, sans-serif",
      "fontSize": "1rem",
    },
    "dark": Object {
      "states": Object {
        "invalid": Object {
          "borderColor": "#FF8A65",
          "color": "#FF8A65",
        },
        "valid": Object {
          "active": Object {
            "borderColor": "#42A5F5",
          },
          "disabled": Object {
            "borderColor": "rgba(206,212,218,0.2)",
            "color": "rgba(255,255,255,0.2)",
          },
          "normal": Object {
            "borderColor": "#ced4da",
            "color": "#fff",
          },
        },
      },
    },
    "light": Object {
      "states": Object {
        "invalid": Object {
          "borderColor": "#F4511E",
          "color": "#F4511E",
        },
        "valid": Object {
          "active": Object {
            "borderColor": "#1E88E5",
          },
          "disabled": Object {
            "backgroundColor": "#f8f9fa",
            "color": "#868e96",
          },
          "normal": Object {
            "borderColor": "#ced4da",
            "color": "#495057",
          },
        },
      },
    },
  },
  "toggle": Object {
    "states": Object {
      "checked": Object {
        "backgroundColor": "#1E88E5",
      },
      "normal": Object {
        "backgroundColor": "#ced4da",
      },
    },
  },
}
`;
